components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -329,12 +345,15 @@ public void deleteUser(String userId) throws NotFoundException, CharonException
      * @return User list with detailed attributes
      * @throws CharonException Error while listing users
      */
-    private List<Object> listUsers(Map<String, Boolean> requiredAttributes, int offset, int limit, String sortBy,
-            String sortOrder, String domainName) throws CharonException {
+    private List<Object> listUsers(Map<String, Boolean> requiredAttributes, int offset, Integer limit,
+            String sortBy, String sortOrder, String domainName) throws CharonException {
 
         List<Object> users = new ArrayList<>();
-        //0th index is to store total number of results.
+        // 0th index is to store total number of results.
         users.add(0);
+
+        // Handle limit equals NULL scenario.
+        limit = handleLimitEqualsNULL(limit);
```

Comment:
From where will it handle limit < 0 scenario?

Reply:
It is handled from Charon ResourceManagerUtil.processCount(Integer) method